design-documents/catalog-domain/product-data-object.md

```diff
@@ -0,0 +1,377 @@
+# Product schema 
+
+Product entity described with a pseudo schema.
+The purpose of this schema to determine the bare minimum set of data required for rendering all storefront operations.
+
+## Product identity
+
+```json
+{
+  "type": "Product",
+  "fields": [
+    {"name": "id", "type": "string"},
+    {"name": "sku", "type": "string"}
+  ]
+}
+```
+* `id` is an immutable identifier that represent a product in unique way. 
+* `sku` - (stock keeping unit) product natural identifier.
+
+## Product meta information
+
+Meta information defines product structure.
+```json
+{
+  "type": "Product",
+  "fields": [
+    {"name": "type", "type": "ProductType"},
+    {"name": "attributeSet", "type": "string"} 
+  ]
+}
+```
+
+```json
+{
+  "enum": "ProductType",
+  "values": [
+    "simple",
+    "virtual",
+    "downloadable",
+    "giftcard",
+    "grouped",
+    "configurable",
+    "bundle"
+  ]
+}
+```
+
+* `type` - product type defines type specific attributes and options that can be associated with a product.
+* `attributeSet` attribute set name.
+Attribute set declares schema for product dynamic attributes.
+For storefront scenarios attribute set may may specify more precise product taxonomy.
+
+
+## System attributes
+
+Product has a set of default out of the box attributes.
+This set present across all system installations.
+```json
+{
+  "type": "Product",
+  "fields": [
+    {"name": "name", "type": "string"},
+    {"name": "description", "type": "string"},
+    {"name": "shortDescription", "type": "string"},
+    {"name": "weight", "type": "float"},
+    {"name": "weightUnit", "type": "string"},
+    {"name": "visibility", "type": "Visibility"}
+  ]
+}
+```
+```json
+{
+  "enum": "Visibility",
+  "values": ["Not Visible Individually", "Catalog", "Search", "Catalog & Search"]
+}
+```
+* `name` product name.
+* `description` product description.
+* `shortDescription` product short description.
+* `weight` - weight of physical item, makes sense for *Simple* products only. 
+* `weightUnit` - unit for measurements for product weight 
+* `visibility` - tells how the product can be shown if enabled.
+
+
+## Dynamic/EAV Attributes
+
+Product may have user-defined custom attributes.
+Set of attributes may vary and depends on an `attributeSet` assigned to a product.
+
+```json
+{
+  "type": "Product",
+  "fields": [
+    {"name": "attributes", "type": "Attribute", "repeated": "true"}
```

Comment:
What is "repeated"? As I understand it means it's a list/array. Could you please add this explanation somewhere in the document?